lib/install/management/appliance.go

```diff
@@ -93,6 +77,35 @@ func (d *Dispatcher) removeApplianceIfForced(conf *metadata.VirtualContainerHost
 	return nil
 }
 
+func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine) error {
+	var err error
+	// ignore power off error cause VM can be powered off already
+	if _, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
```

Comment:
This is not a fast path, so let's be neat and check the VM power state to avoid errors in the Task view of UIs as much as possible. Not needed for this PR.